indica/models/diode_filters.py

```diff
@@ -63,6 +64,70 @@ def __init__(
         self.calibration = calibration
         self.instrument_method = instrument_method
         self.quantities = AVAILABLE_QUANTITIES[self.instrument_method]
+        self.channel_mask = channel_mask
+
+        wavelength = np.linspace(
+            self.filter_wavelength - self.filter_fwhm * 2,
+            self.filter_wavelength + self.filter_fwhm * 2,
+        )
+        self.wavelength = DataArray(wavelength, coords=[("wavelength", wavelength)])
+
+        # Transmission filter function
+        transmission = ph.make_window(
+            wavelength,
+            self.filter_wavelength,
+            self.filter_fwhm,
+            window=self.filter_type,
+        )
+        self.transmission = DataArray(transmission, coords=[("wavelength", wavelength)])
+
+    def integrate_spectra(self, spectra: DataArray, fit_background: bool = True):
+        """
+        Apply the diode transmission function to an input spectra
+
+        Parameters
+        ----------
+        spectra
+            Spectra with dimensions (channel, wavelength, time) in any order,
+            and with units of W/m**2
+        fit_background
+            If True - background fitted and then integrated
+            If False - spectra integrated using filter without any fitting
+
+        Returns
+        -------
+        Background emission & integral of spectra using the filter transmission
+        TODO: uncertainty on fit not calculated
+        """
+
+        # Interpolate transmission filter on spectral wavelength & restrict to > 0
+        _transmission = self.transmission.interp(wavelength=spectra.wavelength)
+        transmission = _transmission.where(_transmission > 1.0e-3, drop=True)
+        wavelength_slice = slice(
+            np.min(transmission.wavelength), np.max(transmission.wavelength)
+        )
+
+        # Take away neutron spikes in pixel intensity
+        _spectra = spectra.sortby("wavelength").sel(wavelength=wavelength_slice)
+        _spectra = _spectra.where(
+            xr.ufuncs.fabs(_spectra.diff("wavelength", n=2)) < 0.4e-3
+        )
+
+        # Fit spectra to calculate background emission, filter and integrate
+        if fit_background:
```

Comment:
This should probably be it's own function

Reply:
Absolutely! Where would you put it? Shall we make a new file in indica/operators/ for spectroscopy methods? 

I'll add a TODO